src/plans/components/ProviderSelect.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { ComponentProps, FC, ReactNode } from 'react';
+import { getResourceUrl } from 'src/modules/Providers';
+
+import { ExternalLink } from '@components/common/ExternalLink';
+import Select from '@components/common/Select';
+import { ProviderModelGroupVersionKind, ProviderModelRef, V1beta1Provider } from '@kubev2v/types';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { EmptyState, EmptyStateVariant, SelectOption } from '@patternfly/react-core';
+import { ForkliftTrans, useForkliftTranslation } from '@utils/i18n';
+
+type ProviderSelectProps = Pick<ComponentProps<typeof Select>, 'onSelect' | 'status'> & {
+  id: string;
+  value: string;
+  namespace: string;
+  placeholder?: string;
+  emptyState?: ReactNode;
+};
+
+export const ProviderSelect: FC<ProviderSelectProps> = ({
+  id,
+  value,
+  namespace,
+  placeholder,
+  emptyState,
+  status,
+  onSelect,
+}) => {
+  const { t } = useForkliftTranslation();
+  const [providers, providersLoaded, providerError] = useK8sWatchResource<V1beta1Provider[]>({
+    namespace,
+    groupVersionKind: ProviderModelGroupVersionKind,
+    namespaced: true,
+    isList: true,
+  });
+
+  const providersListUrl = getResourceUrl({
+    reference: ProviderModelRef,
+    namespace: namespace,
+  });
+
+  return (
+    <Select
+      id={id}
+      value={value}
+      status={status}
+      onSelect={onSelect}
+      placeholder={
+        placeholder ||
+        (!providersLoaded || providerError ? t('Providers failed to load.') : undefined)
+      }
+      isDisabled={!providersLoaded || providerError}
+    >
+      {providers.length === 0 &&
+        (emptyState || (
+          <EmptyState variant={EmptyStateVariant.sm}>
+            <ForkliftTrans>
+              There are no providers in project <strong>{namespace}</strong>. Create one from the{' '}
+              <ExternalLink href={providersListUrl} isInline>
+                Providers page
+              </ExternalLink>
+            </ForkliftTrans>
+          </EmptyState>
```

Comment:
please extract it to a new component , this can be the start of generic empty state

Reply:
A generic component just for Select empty states or all empty states? I think what's inside the `ForkliftTrans` component will be unique per instance (regardless if scoped for Selects or all empty states), and the variant will differ as well, so the shared component would be very similar to the implementation we see here, except it could have the translation built-in.

Does that seem like enough of a reason for a new component?

This is what it would end up looking like roughly, unless you had something else in mind:

```
export const EmptyState: FC<EmptyStateProps> = ({ children, ...props }) => (
  <PfEmptyState {...props}>
    <ForkliftTrans>{children}</ForkliftTrans>
  </PfEmptyState>
);
```

Reply:
I've made this component, but I'm not sure how useful it is.